### PR TITLE
feat(device): add API for getting and setting timers

### PR DESCRIPTION
**Pull request by arokor**
Adds an api for getting and setting timers on the machine-interface

### CODE REVIEW

File: packages/device/src/interfaces/machine.ts [hunk 1]
```diff
@@ -10,6 +10,42 @@ export interface DeviceInfo {
   userAgent: string; // The user agent string
 }
 
+/**
+ * Possible values for the weekDays property of the TimerEntry interface. "all" means
+ * that the timer should be active every day of the week.
+ */
+export type WeekDay =
```
Comment (manast) [inline packages/device/src/interfaces/machine.ts:17]: wouldn't it be more convenient to use an enum instead?

**Review by manast — APPROVED** [review]

File: packages/device/src/interfaces/machine.ts (hunk 1 above)
Comment (arokor) [reply packages/device/src/interfaces/machine.ts:17]: I find union types to be easier to work with (and during debugging) so I usually use that when there is no specific reason for an enum.
https://www.bam.tech/article/should-you-use-enums-or-union-types-in-typescript

Comment (manast) [reply packages/device/src/interfaces/machine.ts:17]: For me it has 2 major advantages, 1), as soon as you see the code using the enum you understand it is an enum, if you just see "wed" it looks like a string. 2) easier to get autocompletion by just writing the enumtype and . you get the list to choose from.

Comment (manast) [reply packages/device/src/interfaces/machine.ts:17]: But after reading the article, it seems like my arguments are not so good as you can do the same with union types (WeekDay.wed), but maybe we should rename the string to be capitalised then? I think we should settle for some standard and use it everywhere.

Comment (manast) [reply packages/device/src/interfaces/machine.ts:17]: The answer to this question seems more complete: https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant

Comment (arokor) [reply packages/device/src/interfaces/machine.ts:17]: Switched from enums to union types in the machine- and cache interfaces

**Pull request merged**